lib/src/local_working_copy.rs

```diff
@@ -1449,6 +1493,11 @@ impl FileSnapshotter<'_> {
                 repo_path,
                 &content,
                 self.conflict_marker_style,
+                new_file_state
```

Comment:
nit: I think the marker length can be obtained from the current file state, not from the new file state.